scripts/illumina_run_parameter_parser.py

```diff
@@ -91,6 +93,76 @@ def attach_xml(process, run_dir):
+                stats.update({"percent_aligned"       :  getattr(summary.at(read).at(lane), "percent_aligned")().mean()})
+                stats.update({"percent_gt_q30"        :  getattr(summary.at(read).at(lane), "percent_gt_q30")()})
+                stats.update({"percent_pf"            :  getattr(summary.at(read).at(lane), "percent_pf")().mean()})
+                stats.update({"phasing"               :  getattr(summary.at(read).at(lane), "phasing")().mean()})
+                stats.update({"prephasing"            :  getattr(summary.at(read).at(lane), "prephasing")().mean()})
+                stats.update({"reads_pf"              :  getattr(summary.at(read).at(lane), "reads_pf")()})
+                stats.update({"yield_g"               :  getattr(summary.at(read).at(lane), "yield_g")()})
+                if lane_nbr not in run_stats_summary.keys():
+                    run_stats_summary.update({lane_nbr: {read: stats}})
+                else:
+                    run_stats_summary[lane_nbr].update({read : stats})
+    return run_stats_summary
+
+
+def set_run_stats_in_lims(process, run_stats_summary):
+    for art in process.all_outputs():
+        if 'Lane' in art.name:
+            lane_nbr = int(art.name.split(' ')[1])
+            read = 1
+            for i in run_stats_summary[lane_nbr].keys():
+                stats = run_stats_summary[lane_nbr][i]
+                if not math.isnan(stats['density']):
+                    art.udf["Cluster Density (K/mm^2) R{}".format(read)]  =  stats['density']
+                if not math.isnan(stats['error_rate']):
+                    art.udf["% Error Rate R{}".format(read)]              =  stats['error_rate']
+                if not math.isnan(stats['first_cycle_intensity']):
+                    art.udf["Intensity Cycle 1 R{}".format(read)]         =  stats['first_cycle_intensity']
+                if not math.isnan(stats['percent_aligned']):
+                    art.udf["% Aligned R{}".format(read)]                 =  stats['percent_aligned']
+                if not math.isnan(stats['percent_gt_q30']):
+                    art.udf["% Bases >=Q30 R{}".format(read)]             =  stats['percent_gt_q30']
+                if not math.isnan(stats['percent_pf']):
+                    art.udf["%PF R{}".format(read)]                       =  stats['percent_pf']
+                if not math.isnan(stats['phasing']):
+                    art.udf["% Phasing R{}".format(read)]                 =  stats['phasing']
+                if not math.isnan(stats['prephasing']):
+                    art.udf["% Prephasing R{}".format(read)]              =  stats['prephasing']
+                if not math.isnan(stats['reads_pf']):
+                    art.udf["Reads PF (M) R{}".format(read)]              =  stats['reads_pf']/1000000
+                if not math.isnan(stats['yield_g']):
+                    art.udf["Yield PF (Gb) R{}".format(read)]             =  stats['yield_g']
+                read += 1
```

Comment:
Will there always be as many reads as keys in `run_stats_summary[lane_nbr].keys()`?

Reply:
No, there can be 1 or 2 reads. If there is only 1 read `read += 1` will be run but it doesn't affect anything.

Reply:
hmm ok